src/mapping/config/MappingConfiguration.cpp

```diff
@@ -195,10 +204,25 @@ MappingConfiguration::MappingConfiguration(
                                .setDocumentation("Sets kind of preallocation for PETSc RBF implementation")
                                .setOptions({PREALLOCATION_ESTIMATE, PREALLOCATION_COMPUTE, PREALLOCATION_OFF, PREALLOCATION_SAVE, PREALLOCATION_TREE});
 
+  auto attrExecutor = makeXMLAttribute(ATTR_EXECUTOR, "reference-executor")
+                          .setDocumentation("Specifies the execution backend used by Ginkgo.")
+                          .setOptions({"reference-executor", "omp-executor", "cuda-executor", "hip-executor"});
+
+  auto attrSolver = makeXMLAttribute(ATTR_SOLVER, "cg-solver")
+                        .setDocumentation("Specifies the iterative solver used by Ginkgo.")
+                        .setOptions({"cg-solver", "gmres-solver", "mg-solver"});
+
+  auto attrPreconditioner = makeXMLAttribute(ATTR_PRECONDITIONER, "jacobi-preconditioner")
```

Comment:
I guess we also need a way to handle the GPU index. Let's discuss in person here.

Reply:
Yes (as mentioned above). We should enable the user to provide the GPU device id (which always starts with 0). We can just hand this number over to the executor instantiation then.

Reply:
I added a XML tag which enables the user to provide a GPU ID.